### PR TITLE
Switch F# grammar to ionide-fsgrammar

### PR DESCRIPTION
**Pull request by am11**
## Description
Existing submodule for F# https://github.com/fsprojects/atom-fsharp now redirects to a deprecated repository, which suggests to use VSCode's grammar source, i.e. https://github.com/ionide/ionide-fsgrammar.

This was discussed in https://github.com/github/linguist/issues/4702.

While at it, I have also fixed a typo in `add-grammar` script; `--verbose` is not an option instead `--quiet` is.

## Checklist:
- [x] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=source.fsharp&grammar_format=auto&grammar_url=&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fdotnet%2Ffsharp%2Fblob%2Fec5bad3a391357e03ff2286a264f0e4faf7d840d%2Fsrc%2Ffsharp%2FLanguageFeatures.fs&code=
  - New: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fionide%2Fionide-fsgrammar%2Fmaster%2Fgrammar%2Ffsharp.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fdotnet%2Ffsharp%2Fblob%2Fec5bad3a391357e03ff2286a264f0e4faf7d840d%2Fsrc%2Ffsharp%2FLanguageFeatures.fs&code=

### CODE REVIEW

**Comment by cartermp** [comment]
Thanks @am11!

File: tools/grammars/compiler/data.go [hunk 1]
```diff
@@ -46,6 +48,7 @@ var GrammarsInNonStdPath = map[string]bool{
 	"hy.tmLanguage":           true,
 	"abl-tmlanguage":          true,
 	"avro.tmLanguage":         true,
+	"grammar":                 true,
```
Comment (lildude) [inline tools/grammars/compiler/data.go:51]: This should be the name of the grammar as it appears in the filesystem:

```suggestion
	"ionide-fsgrammar":                 true,
```

Comment (am11) [reply tools/grammars/compiler/data.go:51]: I tried this. but it is a bit more complicated. Here we are looking for the the name of (sub)directory inside the submodule, which has name other than `grammars/` (plural). I have sent a PR upstream, so subsequently we could remove this workaround: https://github.com/ionide/ionide-fsgrammar/pull/156.

Comment (am11) [reply tools/grammars/compiler/data.go:51]: `GrammarsInNonStdPath` is used here: https://github.com/github/linguist/blob/5ff272d46e519914ed823d2cc6f670b205a3ea5d/tools/grammars/compiler/loader.go#L125 the map needs the name of the directory on the filesystem, that has the grammar file. in this case name of the directory is `grammar/`.

Comment (am11) [reply tools/grammars/compiler/data.go]: Upstream PR (that moves grammer files from `grammar/` to `grammars/`) has been merged. I have updated the submodule and rebased.

**Review by Alhadis — APPROVED** [review]
LGTM. 👍

**Review by lildude — APPROVED** [review]

**Pull request merged**